rust/src/mysql/mysql.rs

```diff
@@ -0,0 +1,1133 @@
+            return std::ptr::null_mut();
+        }
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_mysql_tx_get_alstate_progress(
+    tx: *mut std::os::raw::c_void, _direction: u8,
+) -> std::os::raw::c_int {
+    let tx = cast_pointer!(tx, MysqlTransaction);
+    if tx.complete {
+        return 1;
+    }
+    return 0;
+}
+
+export_tx_data_get!(rs_mysql_get_tx_data, MysqlTransaction);
+export_state_data_get!(rs_mysql_get_state_data, MysqlState);
+
+/// Get the mysql query
+#[no_mangle]
+pub unsafe extern "C" fn SCMysqlTxGetCommandName(
+    tx: &mut MysqlTransaction, buf: *mut *const u8, len: *mut u32,
+) -> bool {
+    if let Some(command) = &tx.command {
+        if !command.is_empty() {
+            *buf = command.as_ptr();
+            *len = command.len() as u32;
+            return true;
+        }
+    }
+
+    false
+}
+
+/// Get the mysql rows at index i
+#[no_mangle]
+pub unsafe extern "C" fn SCMysqlGetRowsData(
+    tx: &mut MysqlTransaction, i: u32, buf: *mut *const u8, len: *mut u32,
+) -> bool {
+    if let Some(rows) = &tx.rows {
+        if !rows.is_empty() {
+            let index = i as usize;
+            if let Some(row) = rows.get(index) {
+                if !row.is_empty() {
+                    *buf = row.as_ptr();
+                    *len = row.len() as u32;
+                    return true;
+                }
+            }
+        }
+    }
+
+    false
+}
+
+// Parser name as a C style string.
+const PARSER_NAME: &[u8] = b"mysql\0";
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_mysql_register_parser() {
+    let default_port = CString::new("[3306]").unwrap();
+    let mut stream_depth = MYSQL_CONFIG_DEFAULT_STREAM_DEPTH;
+    let parser = RustParser {
+        name: PARSER_NAME.as_ptr() as *const std::os::raw::c_char,
+        default_port: default_port.as_ptr(),
+        ipproto: IPPROTO_TCP,
+        probe_ts: Some(rs_mysql_probing_ts),
+        probe_tc: Some(rs_mysql_probing_tc),
+        min_depth: 0,
+        max_depth: 16,
+        state_new: rs_mysql_state_new,
+        state_free: rs_mysql_state_free,
+        tx_free: rs_mysql_state_tx_free,
+        parse_ts: rs_mysql_parse_request,
+        parse_tc: rs_mysql_parse_response,
+        get_tx_count: rs_mysql_state_get_tx_count,
+        get_tx: rs_mysql_state_get_tx,
+        tx_comp_st_ts: 1,
+        tx_comp_st_tc: 1,
+        tx_get_progress: rs_mysql_tx_get_alstate_progress,
+        get_eventinfo: Some(MysqlEvent::get_event_info),
+        get_eventinfo_byid: Some(MysqlEvent::get_event_info_by_id),
+        localstorage_new: None,
+        localstorage_free: None,
+        get_tx_files: None,
+        get_tx_iterator: Some(
+            crate::applayer::state_get_tx_iterator::<MysqlState, MysqlTransaction>,
+        ),
+        get_tx_data: rs_mysql_get_tx_data,
+        get_state_data: rs_mysql_get_state_data,
+        apply_tx_config: None,
+        flags: APP_LAYER_PARSER_OPT_ACCEPT_GAPS,
+        get_frame_id_by_name: None,
```

Comment:
Are you planning on adding frames support ?

Reply:
No

Reply:
Fixed, please check again.